modules/programs/atuin.nix

```diff
@@ -94,49 +95,117 @@ in {
-          mkdir $atuin_cache
-        }
-        ${
-          lib.getExe cfg.package
-        } init nu ${flagsStr} | save --force ${config.xdg.cacheHome}/atuin/init.nu
+      programs.zsh.initExtra = mkIf cfg.enableZshIntegration ''
+        if [[ $options[zle] = on ]]; then
+          eval "$(${lib.getExe cfg.package} init zsh ${flagsStr})"
+        fi
       '';
-      extraConfig = ''
-        source ${config.xdg.cacheHome}/atuin/init.nu
+
+      programs.fish.interactiveShellInit = mkIf cfg.enableFishIntegration ''
+        ${lib.getExe cfg.package} init fish ${flagsStr} | source
       '';
-    };
-  };
+
+      programs.nushell = mkIf cfg.enableNushellIntegration {
+        extraEnv = ''
+          let atuin_cache = "${config.xdg.cacheHome}/atuin"
+          if not ($atuin_cache | path exists) {
+            mkdir $atuin_cache
+          }
+          ${
+            lib.getExe cfg.package
+          } init nu ${flagsStr} | save --force ${config.xdg.cacheHome}/atuin/init.nu
+        '';
+        extraConfig = ''
+          source ${config.xdg.cacheHome}/atuin/init.nu
+        '';
+      };
+    }
+
+    (mkIf cfg.daemon.enable (mkMerge [
+      {
+        assertions = [
+          {
+            assertion = versionAtLeast cfg.package.version "18.2.0";
+            message = ''
+              The atuin daemon requires at least version 18.2.0 or later.
+            '';
+          }
+          {
+            assertion = isLinux || isDarwin;
+            message =
+              "The atuin daemon can only be configured on either Linux or macOS.";
+          }
+        ];
+
+        programs.atuin.settings = { daemon = { enabled = true; }; };
+      }
+      (mkIf isLinux {
+        programs.atuin.settings = { daemon = { systemd_socket = true; }; };
+
+        systemd.user.services.atuin-daemon = {
+          Unit = {
+            Description = "atuin daemon";
+            Requires = [ "atuin-daemon.socket" ];
+          };
+          Install = {
+            Also = [ "atuin-daemon.socket" ];
+            WantedBy = [ "default.target" ];
+          };
+          Service = {
+            ExecStart = "${lib.getExe cfg.package} daemon";
+            Environment = [ "ATUIN_LOG=info" ];
+            Restart = "on-failure";
+            RestartSteps = 3;
+            RestartMaxDelaySec = 6;
+          };
+        };
+
+        systemd.user.sockets.atuin-daemon = {
+          Unit = { Description = "atuin daemon socket"; };
+          Install = { WantedBy = [ "sockets.target" ]; };
+          Socket = {
+            ListenStream = "%h/.local/share/atuin/atuin.sock";
+            SocketMode = "0600";
+            RemoveOnStop = true;
+          };
+        };
+      })
+      (mkIf isDarwin {
+        launchd.agents.atuin-daemon = {
+          enable = true;
+          config = {
+            ProgramArguments = [ "${lib.getExe cfg.package}" "daemon" ];
+            EnvironmentVariables = { ATUIN_LOG = "info"; };
```

Comment:
I would add a cfg key behind lib.optionals for this - people might not like Atuin's daemon being too chatty.